src/Catrobat/AppBundle/Listeners/DownloadStatisticsListener.php

```diff
@@ -9,29 +9,61 @@
 class DownloadStatisticsListener
 {
     private $download_statistics_service;
+    private $security_token_storage;
 
-    public function __construct($download_statistics_service)
+    public function __construct($download_statistics_service, $security_token_storage)
     {
         $this->download_statistics_service = $download_statistics_service;
+        $this->security_token_storage = $security_token_storage;
     }
-    
+
     public function onTerminateEvent(PostResponseEvent $event)
     {
         $attributes = $event->getRequest()->attributes;
         if ($attributes->has('download_statistics_program_id')) {
             $program_id = $attributes->get('download_statistics_program_id');
-            $ip = $event->getRequest()->server->get('REMOTE_ADDR');
-            $user_agent = $event->getRequest()->headers->get('User-Agent') ;
-            $this->createProgramDownloadStatistics($program_id, $ip, $user_agent);
+            $ip = $this->getIp($event);
```

Comment:
Why not just use [getClientIp from Request](http://api.symfony.com/2.0/Symfony/Component/HttpFoundation/Request.html#method_getClientIp)?